Packs/MicrosoftDefenderAdvancedThreatProtection/Integrations/MicrosoftDefenderAdvancedThreatProtection/MicrosoftDefenderAdvancedThreatProtection.py

```diff
@@ -2559,7 +2576,273 @@ def get_indicator_dbot_object(indicator):
     return get_dbot_indicator(indicator_type, dbot, indicator_value)
 
 
-''' EXECUTION CODE '''
+''' LIVE RESPONSE CODE '''
+
+
+def run_polling_command(client: MsClient, args: dict, cmd: str, action_func: Callable,
+                        results_function: Callable, post_polling_process: Callable):
+    """
+    This function is generically handling the polling flow. In the polling flow, there is always an initial call that
+    starts the uploading to the API (referred here as the 'upload' function) and another call that retrieves the status
+    of that upload (referred here as the 'results' function).
+    The run_polling_command function runs the 'upload' function and returns a ScheduledCommand object that schedules
+    the next 'results' function, until the polling is complete.
+    Args:
+        args: the arguments required to the command being called, under cmd
+        cmd: the command to schedule by after the current command
+        upload_function: the function that initiates the uploading to the API
+        results_function: the function that retrieves the status of the previously initiated upload process
+        uploaded_item: the type of item being uploaded
+
+    Returns:
+
+    """
+    ScheduledCommand.raise_error_if_not_supported()
+    interval_in_secs = int(args.get('interval_in_seconds', 5))
+    # distinguish between the initial run, which is the upload run, and the results run
+    is_first_run = 'machine_action_id' not in args
+    demisto.debug(f'polling args: {args}')
+    if is_first_run:
+        command_results = action_func(client, args)
+        outputs = command_results.outputs
+        # schedule next poll
+        polling_args = {
+            'machine_action_id': outputs.get('action_id'),
+            'interval_in_seconds': interval_in_secs,
+            'polling': True,
+            **args,
+        }
+        scheduled_command = ScheduledCommand(
+            command=cmd,
+            next_run_in_seconds=interval_in_secs,
+            args=polling_args,
+            timeout_in_seconds=600)
+        command_results.scheduled_command = scheduled_command
+        return command_results
+
+    # not a first run
+
+    command_result = results_function(client, args)
+    action_status = command_result.outputs.get("status")
+    command_status = command_result.outputs.get("commands", [])[0].get("commandStatus")
+    if action_status == 'Failed' or command_status == 'Failed':
+        raise Exception(f'Command failed to get results. {command_result.outputs.get("commands", [])[0].get("errors")}')
```

Comment:
Getting `[0]` of empty `list` will still raise an error. either wrap it with an `if` condition or provide another `default` value.